### ISSUE TITLE
Maven pom.xml EOF newline getting removed

### ISSUE THREAD

**Issue by jpalomaki** [googleapis/release-please-action#743]
### TL;DR

Release-please generated PRs that update `<version />` in a maven project `pom.xml` remove the EOF newline

### Expected behavior

I'd expect the EOF newline to be retained

### Observed behavior

EOF newline is removed from `pom.xml`, and GitHub highlights it with red symbol
![Screenshot 2023-03-23 at 16 05 01](https://user-images.githubusercontent.com/582404/227228865-b682a228-4f48-4af3-b8e8-c652a666f156.png)

### Action YAML

```yaml
name: release-please

on:
  push:
    branches:
      - master

jobs:
  this:
    runs-on: ubuntu-22.04
    steps:

      # Obtain a GitHub app installation token (app has necessary permissions to this repo)
      - id: auth
        uses: ...
        with:
          ...

      - uses: google-github-actions/release-please-action@v3
        with:
          release-type: maven
          package-name: my-package
          token: ${{ steps.auth.outputs.token }}
```

### Additional information

We use a GitHub app installation token, because we need to trigger other workflows on `release` events.

However, I don't think this configuration has anything to do with the observed bug, as everything else works fine.

**Comment by chingor13** [comment]
This would need to be fixed on the upstream library: https://github.com/googleapis/release-please

**Comment by jpalomaki** [comment]
> This would need to be fixed on the upstream library: https://github.com/googleapis/release-please

@chingor13 Do I need to create a separate issue there?

**Comment by chingor13** [comment]
Yes, it cannot be fixed here.

**Comment by jpalomaki** [comment]
> Yes, it cannot be fixed here.

Created upstream issue here: https://github.com/googleapis/release-please/issues/1894

**Comment by chingor13** [comment]
This was fixed in 15.10.1 and will be included in #757
